### DISCUSSION TITLE
Mozc is disabled on browsers

### DISCUSSION THREAD

**Discussion by Ks4four** [google/mozc#805] [Q&A] [answered]
## Description
Mozc didn't work on browsers in my system.

## Steps to reproduce
1. Go to Firefox, Chrome, Etc...
2. Click on any input boxes
3. Type anything

## Expected behavior
It should be able to convert or have a suggestion box below my cursor.

## Actual behavior
It's unable to reach the IME options, even if it says "Right click to show IME options".
It also failed to suggest and let the text box making direct input.
It can't change to katakana mode, hiragana mode, or direct input... etc.
It seems like it's disabled.

## Screenshots
![image](https://github.com/google/mozc/assets/78544646/ab8b9ff2-e296-4ca3-b494-bcdc59d27001)

## Version or commit-id
Mozc-2.29.5210.100+24.11.oss

## Environment
 - OS: Windows 10 Pro 22H2, 19045.3393
 - Firefox Developer Edition, Chrome, Microsoft Edge

## Investigations
 - This issue doesn't happen on Google Japanese IME.

 - What applications this issue happens on.
   + Firefox Developer Edition (x64 en-US) 118.0
   + Chrome 116.0.5845.180, via Scoop
   + Microsoft Edge 116.0.1938.69

 - What applications this issue does not happen on.
   + anything that not a browser by far

 - (optional) What versions or commit-ids this issue did not happen on
   + no-investigation

## Additional context
Setting region and language to Japan and Japanese doesn't solve the problem.

**Comment by yukawa** [comment, marked as answer]
Hmm, I cannot reproduce the issue locally.  If your issue is specific to web browsers, once thing I suspect is that your environment has some antivirus software that prevents Mozc DLLs (e.g. `mozc_tip64.dll`) from establishing a named-pipe connection to `mozc_server.exe`.

If you happen to be familiar with Windows software development, one of the most straightforward ways to figure out what's going on would be debugging the following lines in the browser process (e.g. `chrome.exe`).
https://github.com/google/mozc/blob/f79e092ac7068f6bbdacb2b069641771e7cb1f43/src/ipc/win32_ipc.cc#L636-L752

(Another possibility is that `mozc_tip64.dll` is not loaded at all into the browser process)

**Comment by Ks4four** [comment]
Thanks for the response and the information about antivirus software.
It is founded that ESET blocking the mozc_tip64.dll for connections, and it works if the Banking & Payment protection was disabled.
I think it's possible to make a whitelist for mozc.
![image](https://github.com/google/mozc/assets/78544646/08fefd54-13e0-4164-8b21-a6767982ff28)
